lib/cursor.js

```diff
@@ -634,10 +635,8 @@ var nextFunction = function(self, callback) {
     // Set as init
     self.cursorState.init = true;
 
-    // Server does not support server
-    if (self.cmd && self.cmd.collation && self.server.ismaster.maxWireVersion < 5) {
-      return callback(new MongoError(f('server %s does not support collation', self.server.name)));
-    }
+    // if collation not supported, will call callback with new MongoError
+    if (collationNotSupported(self.server, self.cmd, callback)) return;
```

Comment:
@rweinberger I thought the idea was to call the callback in place, instead of passing it to the helper function:
```
    if (collationNotSupported(self.server, self.cmd)) {
      return callback(new MongoError(f('server %s does not support collation', self.server.name)));
    }
```

Reply:
Right, I haven't actually updated it from the original design yet since I was waiting for more input into the design issue, but if this is design good with everyone I can go ahead and change it @daprahamian @kvwalker

Reply:
I am on board with the changes. 👍

Reply:
Me too!